fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaClient.java

```diff
@@ -403,6 +388,86 @@ private List<Partition> getPartitionsWithRetry(String dbName, String tableName,
+                throw connectionException;
+            } finally {
+                if (client == null && connectionException != null) {
+                    LOG.error("Failed to get hive client. {}", connectionException.getMessage());
+                } else if (connectionException != null) {
+                    LOG.error("An exception occurred when using the current long link " +
+                            "to access metastore. msg: {}", connectionException.getMessage());
+                    client.close();
+                } else if (client != null) {
+                    client.finish();
+                }
+            }
+        }
+        return partitionStats;
+    }
+
+    private Map<String, List<ColumnStatisticsObj>> getPartitionColumnStatsWithRetry(String dbName,
+                                                                                    String tableName,
+                                                                                    List<String> partNames,
+                                                                                    List<String> columnNames,
+                                                                                    int retryNum)
+            throws StarRocksConnectorException {
+        int subListSize = (int) Math.pow(2, retryNum);
+        int subListNum = partNames.size() / subListSize;
+        if (subListNum == 0) {
+            subListNum = 1;
+        }
+        List<List<String>> partNamesList = Lists.partition(partNames, subListNum);
+        Map<String, List<ColumnStatisticsObj>> partitionStats = new HashMap<>();
+
+        LOG.warn("Execute getPartitionColumnStatistics on [{}.{}] with {} times retry, slice size is {}, partName size is {}",
+                dbName, tableName, retryNum, subListSize, partNames.size());
+
+        RecyclableClient client = null;
+        try {
+            client = getClient();
+            for (List<String> parts : partNamesList) {
+                partitionStats.putAll(client.hiveClient.getPartitionColumnStatistics(dbName, tableName, parts, columnNames));
+            }
+            LOG.info("Succeed to getPartitionColumnStatistics on [{}.{}] with {} times retry, slice size is {}," +
+                            " partName size is {}", dbName, tableName, retryNum, subListSize, partNames.size());
+            return partitionStats;
+        } catch (TTransportException te) {
+            if (subListNum > 1) {
+                return getPartitionColumnStatsWithRetry(dbName, tableName, partNames, columnNames, retryNum + 1);
+            } else {
+                throw new StarRocksConnectorException("Failed to getPartitionColumnStatistics on [%s.%s] with slice size is %d",
+                        dbName, tableName, subListNum);
+            }
+        } catch (Exception e) {
+            throw new StarRocksConnectorException("Failed to getPartitionColumnStatistics on [%s.%s], msg: %s",
```

Comment:
what's the error message if retrying failed ? looks like it would be a messy error message ?